library/build/tools.pl

```diff
@@ -435,12 +442,25 @@
     has_program(path(CC), _),
     !.
 
+default_cxx_compiler(CXX) :-
+    getenv('CXX', CXX),
+    !.
+default_cxx_compiler(CXX) :-
+    preferred_cxx_compiler(CXX),
+    has_program(path(CXX), _),
+    !.
+
 preferred_c_compiler(CC) :-
     current_prolog_flag(c_cc, CC).
 preferred_c_compiler(gcc).
```

Comment:
See https://github.com/SWI-Prolog/swipl-devel/issues/1294

Changing this to `preferred_c_compiler(path(gcc))` breaks `build_step(configure, ...)`.